ext/nnstreamer/tensor_decoder/tensordec-imagesegment.c

```diff
@@ -0,0 +1,350 @@
+  guint **segment_map;  /**< The image segmentated map */
+
+  /*  From option2 */
+  guint o_width; /**< Output video Width */
+  guint o_height; /**< Output video Height */
+
+  /*  From option3 */
+  guint i_width; /**< Input video Width */
+  guint i_height; /**< Input video Height */
+} image_segments;
+
+/** @brief Initialize image_segments per mode */
+static int
+_init_modes (image_segments * idata)
+{
+  if (idata->mode == TFLITE_DEEPLAB_IMAGE_SEGMENT) {
+    int i;
+
+    idata->i_width = TFLITE_DEEPLAB_IMAGE_WIDTH;
+    idata->i_height = TFLITE_DEEPLAB_IMAGE_HEIGHT;
+    idata->o_width = TFLITE_DEEPLAB_IMAGE_WIDTH;
+    idata->o_height = TFLITE_DEEPLAB_IMAGE_HEIGHT;
+    idata->segment_map = g_new0 (guint *, idata->i_height);
+    for (i = 0; i < idata->i_height; i++) {
+      idata->segment_map[i] = g_new0 (guint, idata->i_width);
+    }
+  }
+
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+is_init (void **pdata)
+{
+  image_segments *idata;
+  *pdata = g_new0 (image_segments, 1);
+
+  idata = *pdata;
+  idata->mode = UNKNOWN_IMAGE_SEGMENT;
+  idata->i_width = 0;
+  idata->i_height = 0;
+  idata->o_width = 0;
+  idata->o_height = 0;
+  idata->segment_map = NULL;
+
+  return TRUE;
+}
+
+/** @brief Free the allocated segment_map */
+static void
+_free_segment_map (image_segments * idata)
+{
+  int i;
+
+  if (idata->segment_map) {
+    for (i = 0; i < idata->i_height; i++) {
+      g_free (idata->segment_map[i]);
+    }
+    g_free (idata->segment_map);
+  }
+
+  idata->segment_map = NULL;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static void
+is_exit (void **pdata)
+{
+  image_segments *idata = *pdata;
+
+  _free_segment_map (idata);
+
+  g_free (*pdata);
+  *pdata = NULL;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+is_setOption (void **pdata, int op_num, const char *param)
+{
+  image_segments *idata = *pdata;
+
+  if (op_num == 0) {
+    /* option1 = image segmentation decoder mode */
+    image_segment_modes previous = idata->mode;
+    idata->mode = find_key_strv (is_modes, param);
+
+    if (NULL == param || *param == '\0') {
+      GST_ERROR ("Please set the valid mode at option1");
+      return FALSE;
+    }
+
+    if (idata->mode != previous && idata->mode != UNKNOWN_IMAGE_SEGMENT) {
+      return _init_modes (idata);
+    }
+    return TRUE;
+  } else if (op_num == 1) {
+    /* option2 = output video size (width:height) */
+    tensor_dim dim;
+    int rank = gst_tensor_parse_dimension (param, dim);
+
+    idata->o_width = 0;
+    idata->o_height = 0;
+    if (param == NULL || *param == '\0')
+      return TRUE;
+
+    if (rank < 2) {
+      GST_ERROR
+          ("mode-option-2 of image segmentation is video output dimension (WIDTH:HEIGHT). "
+          "The given parameter, \"%s\", is not acceptable.", param);
+      return TRUE;              /* Ignore this param */
```

Comment:
TRUE?

Reply:
TRUE means just ignore wrong option and keep going with correct options.